src/components/Link/Link.stories.mdx

```diff
@@ -0,0 +1,232 @@
+import { LinkTypes } from "./LinkTypes";
+import Icon from "../Icons/Icon";
+
+<Meta
+  title="Link"
+  component={Link}
+  decorators={[withDesign]}
+  parameters={{
+    design: {
+      type: "figma",
+      url:
+        "https://www.figma.com/file/qShodlfNCJHb8n03IFyApM/Master?node-id=16115%3A407",
+    },
+  }}
+  argTypes={{
+    blockName: { table: { disable: true } },
+    modifiers: { table: { disable: true } },
+    children: { table: { disable: true } },
+    type: {
+      control: {
+        type: "select",
+      },
+    },
+  }}
+/>
+
+# Link
+
+<Description of={Link} />
+
+<Preview withToolbar>
+  <Story
+    name="Basic"
+    args={{
+      id: "nypl-link",
+      type: LinkTypes.Action,
+      href: "https://nypl.org",
+      className: "custom-class",
+      attributes: {
+        rel: "nofollow",
+        onClick: e => e.preventDefault(),
+      },
+    }}
+  >
+    {args => <Link {...args}>Link</Link>}
+  </Story>
+</Preview>
+
+<ArgsTable story="Basic" />
+
+## Links With Icons
+
+To render an `Icon` as part of the link, make sure that the `Link` component
+wraps the `Icon`. Use `type={LinkTypes.Action}` to apply appropriate styling to
+links with icons. Icons can be rendered to the left or right of the link text.
+
+<Preview>
+  <Story name="With Icons">
+    <>
+      <Link type={LinkTypes.Action} href="#passed-in-link">
+        <Icon
+          name={"headset"}
+          modifiers={["left", "small"]}
+          decorative={true}
+        />
+        Headset Link
+      </Link>
+      <br />
+      <Link type={LinkTypes.Action} href="#passed-in-link">
+        <Icon name={"clock"} modifiers={["left", "small"]} decorative={true} />
+        Clock Link
+      </Link>
+      <br />
+      <Link type={LinkTypes.Action} href="#passed-in-link">
+        <Icon name={"check"} modifiers={["left", "small"]} decorative={true} />
+        Check Link
+      </Link>
+      <br />
+      <Link type={LinkTypes.Action} href="#passed-in-link-right">
+        Check Link Right
+        <Icon name={"check"} modifiers={["right", "small"]} decorative={true} />
+      </Link>
+    </>
+  </Story>
+</Preview>
+
+## Anchor Element Rendering
+
+`Link` can wrap an existing `<a>` tag or it can use the `href` prop to generate
+an `<a>` tag.
+
+<Preview>
+  <Story name="Anchor Element Rendering">
+    <>
+      This is a{" "}
+      <Link type={LinkTypes.Action}>
+        <a href="#existing-anchor-tag">link</a>
+      </Link>{" "}
+      with the &lt;a&gt; tag as a child of the Link component. And this is a{" "}
+      <Link type={LinkTypes.Action} href="#passed-in-link">
+        link
+      </Link>{" "}
+      where the Link component uses the `href` prop and has a string-only child.
+    </>
+  </Story>
+</Preview>
+
+## Link with Routers
+
+<Preview>
+  <Story name="Link with Routers">
+    <div>With React Router</div>
+    <Router>
+      <Link type={LinkTypes.Action}>
+        <ReactRouterLink to="#">Next Page</ReactRouterLink>
```

Comment:
Nit: Perhaps change the placeholder text to "React Router Link" and "Next Link" so that there isn't a chance "Next Page" is confused with the "NextJS"